### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by stuartleeks**
* Add CRD finalizer and support deleting CRD
* Add integration tests for resource group

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,4 +7,8 @@ kind-create:
 	./scripts/init-kind-cluster.sh
 
 terraform-hack-init:
-	./hack/init.sh
\ No newline at end of file
+	./hack/init.sh
+
+integration-tests:
+	# TODO automatically run the operator
+	ginkgo --progress
```
Comment (lawrencegripper) [inline Makefile:14]: Do we need Ginkgo in the devcontainer?

File: integration_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+					"name":     name,
+					"location": "westeurope",
+				},
+			},
+		}
+		options := metav1.CreateOptions{}
+		_, err := k8sClient.Resource(gvrResourceGroup).Namespace("default").Create(context.TODO(), &obj, options)
+		Expect(err).To(BeNil())
+
+		By("returning the resource ID")
```
Comment (lawrencegripper) [inline integration_test.go:49]: Oh nice, I'd not used the `by` func before

File: Makefile (hunk 1 above)
Comment (stuartleeks) [reply Makefile:14]: Oops - missed a commit. Will add that in the next PR!

File: integration_test.go (hunk 2 above)
Comment (stuartleeks) [reply integration_test.go:49]: Yeah, I'm still finding my feet a bit with ginkgo, but `By` let me keep some output showing the flow but control the ordering :-)

**Pull request merged**